src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -94,6 +100,12 @@ internal Task InitializeChannelsAsync()
             return InitializeWebHostRuntimeChannelsAsync();
         }
 
+        private void KillRpcServer(ScriptHostStateChangedEvent scriptHostShutdownEvent)
+        {
+            _logger.LogDebug($"Killing RPC server due to ScriptHostState change from {scriptHostShutdownEvent.OldState.ToString()} to {scriptHostShutdownEvent.NewState.ToString()}");
+            _rpcServer.KillAsync().Wait();
```

Comment:
Let's discuss improvements to this....

Reply:
This is a good point. @yojagad - Can you try using shutdown instead of kill?

Reply:
@pragnagopa I did consider using shutdown , and tried it out too. The only caveat is, there's still a very small probability that this might still get invoked before the language workers are shut down (ie., if it takes longer than 10 seconds) and shutdown in that case would wait until the language worker channel dispose is complete. That's the reason I left it as KillAsync. We could try attempting a shutdown and if it doesn't happen within a specified duration, do a KillAsync()?

Reply:
@pragnagopa  and @fabiocav  Did a shutdown with timeout followed by Kill and added unit test for that as well